src/main/java/assertions/NewAccountAssertion.java

```diff
@@ -0,0 +1,24 @@
+package assertions;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.testng.Assert;
+import pages.MainPage;
+
+public class NewAccountAssertion extends MainPage{
+
+    @FindBy(xpath = "//div[@id='rightPanel']/p[text()='Congratulations, your account is now open.']")
+    WebElement messageSuccessful;
+
+    public NewAccountAssertion(WebDriver driver) {
+        super(driver);
+        PageFactory.initElements(driver,this);
+    }
+
+    public void hasUserOpenedNewAccount(){
+        Assert.assertTrue(messageSuccessful.getText().contains("Congratulations, your account is now open."));
```

Comment:
Jeśli wyszukujesz element po tekscie to zawsze jeśli go znajdzie to ta asercja przejdzie :) Jeśli miałby się wywalić to by nie znalazł elementu.

Reply:
Szczerze mówiąc trochę nie rozumiem komentarza. Czy ta asercja jest zła? Mam jakiś element ("message successful") i w nim szukam tekstu (konkretnego). No i zgadzam się że jak nie znajdzie elementu to się wywali, ale czy to źle? A w taki razie jakby to trzeba było poprawić?

Reply:
chodzi o to że selector do messageSuccessful ma w sobie wyszukiwanie po tekście.  Jeśli w takim razie go wyszuka po tekście jaki zaweira to potem asercja na tym elemencie sprawdzająca czy ma taki tekst jest już trochę duplikowaniem. Lepiej wtedy sprawdzić czy ten element jest displayed bo przecież już wyszukujemy go po tym tekście.